### PR TITLE
First step towards forklifting some go-ipfs configuration functionality into estuary

### PR DESCRIPTION
**Pull request by softwareplumber**
estuary/node is configured by a  Config struct in the node package, which has some function pointers which allow initialization code specific to estuary or estuary-shuttle to be injected.

This PR removes the function pointers from Config and moves it into a config package, where additional functionality (notably for reading/writing a json confif file) will be added.

Function pointers are replaced by an interface defined in node which we implement independently for estuary and estuary-shuttle

### CODE REVIEW

**Review by gmelodie — CHANGES_REQUESTED** [review]
Just one small comment, overall looking good! Thank you for putting this in place!

File: node/node.go [hunk 1]
```diff
@@ -170,7 +148,7 @@ func Setup(ctx context.Context, cfg *Config) (*Node, error) {
 		libp2p.Identity(peerkey),
 		libp2p.BandwidthReporter(bwc),
 		libp2p.DefaultTransports,
-		libp2p.ResourceManager(network.NullResourceManager),
+		libp2p.ResourceManager(rcm),
```
[the remark this answers is not in the document]
Comment (gmelodie) [reply node/node.go:151]: I forgot to highlight the lines above these ones but consider them in this comment as well.

[the remark this answers is not in the document]
Comment (softwareplumber) [reply node/node.go:151]: You're right, this is me rolling back d00e817f93b9657d3be640e078ddf8a545cd8135 so I could do some testing without taking down internet connectivity for my entire company! I'll happily roll back those changes but the resulting build will not be tested locally.

**Comment by softwareplumber** [comment]
Latest push creates a config option 'NoLimiter' which defaults to true; when true, the NullResourceManager is used so functionality should be unchanged by this PR

**Review by jimmylee — COMMENTED** [review]
LGTM, gonna get @whyrusleeping to look at this as well.

**Pull request merged**